models/project_board.go

```diff
@@ -0,0 +1,226 @@
+	ProjectBoardList []*ProjectBoard
+)
+
+const (
+	// None is a project board type that has no predefined columns
+	None ProjectBoardType = iota
+
+	// BasicKanban is a project board type that has basic predefined columns
+	BasicKanban
+
+	// BugTriage is a project board type that has predefined columns suited to hunting down bugs
+	BugTriage
+)
+
+// ProjectBoard is used to represent boards on a project
+type ProjectBoard struct {
+	ID      int64 `xorm:"pk autoincr"`
+	Title   string
+	Default bool //if true it collects issues witch are not signed to a specific board jet
+
+	ProjectID int64 `xorm:"INDEX NOT NULL"`
+	RepoID    int64 `xorm:"INDEX NOT NULL"`
+	CreatorID int64 `xorm:"NOT NULL"`
+
+	CreatedUnix timeutil.TimeStamp `xorm:"INDEX created"`
+	UpdatedUnix timeutil.TimeStamp `xorm:"INDEX updated"`
+
+	Issues []*Issue `xorm:"-"`
+}
+
+// IsProjectBoardTypeValid checks if the project board type is valid
+func IsProjectBoardTypeValid(p ProjectBoardType) bool {
+	switch p {
+	case None, BasicKanban, BugTriage:
+		return true
+	default:
+		return false
+	}
+}
+
+func createBoardsForProjectsType(sess *xorm.Session, project *Project) error {
+
+	var items []string
+
+	switch project.BoardType {
+
+	case BugTriage:
+		items = setting.Repository.ProjectBoardBugTriageType
+
+	case BasicKanban:
+		items = setting.Repository.ProjectBoardBasicKanbanType
+
+	case None:
+		fallthrough
+	default:
+		return nil
+	}
+
+	if len(items) == 0 {
+		return nil
+	}
+
+	var boards = make([]ProjectBoard, 0, len(items))
+
+	for _, v := range items {
+		boards = append(boards, ProjectBoard{
+			CreatedUnix: timeutil.TimeStampNow(),
+			UpdatedUnix: timeutil.TimeStampNow(),
+			CreatorID:   project.CreatorID,
+			Title:       v,
+			ProjectID:   project.ID,
+			RepoID:      project.RepoID,
+		})
+	}
+
+	_, err := sess.Insert(boards)
+	return err
+}
+
+// NewProjectBoard adds a new project board to a given project
+func NewProjectBoard(board *ProjectBoard) error {
+	_, err := x.Insert(board)
+	return err
+}
+
+// DeleteProjectBoardByID removes all issues references to the project board.
+func DeleteProjectBoardByID(repoID, projectID, boardID int64) error {
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return err
+	}
+
+	if err := deleteProjectBoardByID(sess, repoID, projectID, boardID); err != nil {
+		return err
+	}
+
+	return sess.Commit()
+}
+
+func deleteProjectBoardByID(e Engine, repoID, projectID, boardID int64) error {
+	board, err := getProjectBoard(e, repoID, projectID, boardID)
+	if err != nil {
+		if IsErrProjectBoardNotExist(err) {
+			return nil
+		}
+
+		return err
+	}
+
+	if err = board.removeIssues(e); err != nil {
+		return err
+	}
+
+	if _, err := e.ID(board.ID).Delete(board); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetProjectBoard fetches the current board of a project
+func GetProjectBoard(repoID, projectID, boardID int64) (*ProjectBoard, error) {
```

Comment:
Ditto the above for this pair of functions, presumably the projectID should be sufficient if we're looking for "the current board," but I haven't looked at callers to see if this is true.